examples/example-index.js

```diff
@@ -0,0 +1,423 @@
+      indexOfPropertiesAndStates[prop] = [];
+    }
+    indexOfPropertiesAndStates[prop].push(example);
+  }
+}
+
+function addLandmarkRole (landmark, hasLabel, title, ref) {
+  var example = {};
+
+  example.title = title;
+  example.ref = ref;
+  addExampleToRoles(landmark, example);
+  if (hasLabel) {
+    addExampleToPropertiesAndStates('aria-labelledby', example);
+  }
+}
+
+var count = 0;
+
+function findHTMLFiles (path) {
+  fs.readdirSync(path).forEach(function (file) {
+    var newPath = path + '/' + file;
+
+    var stats = fs.statSync(newPath);
+
+    if (stats.isDirectory()) {
+      findHTMLFiles(newPath);
+    }
+
+    if (stats.isFile() &&
+        (newPath.indexOf('.html') > 0) &&
+        (newPath.indexOf('index.html') < 0) &&
+        (newPath.indexOf('landmark') < 0)) {
+      count += 1;
+
+      var data = fs.readFileSync(newPath, 'utf8');
+
+      var ref = newPath;
+      var title = getTitle(data);
+      var roles = getRoles(data);
+      var props = getPropertiesAndStates(data);
+
+      console.log('\nFile ' + count + ': ' + ref);
+      console.log('Title  ' + count + ': ' + title);
+      console.log('Roles  ' + count + ': ' + roles);
+      console.log('Props  ' + count + ': ' + props);
+
+      var example = {};
+
+      example.title = title;
+      example.ref = ref;
+
+      addExampleToRoles(roles, example);
+
+      addExampleToPropertiesAndStates(props, example);
+    }
+  });
+};
+
+findHTMLFiles('.');
+
+var res = fs.readdir('.');
+
+for (r in res) {
+
+  // Handle landmark examples separately
+  if (res[r].indexOf('landmark') < 0) {
+  }
+}
+
+// Add landmark examples, since they are a different format
+
+addLandmarkRole('banner', false, 'Banner Landmark', 'http://localhost/GitHub/aria-practices/examples/landmarks/banner.html');
+addLandmarkRole('complementary', true, 'Complementary Landmark', 'http://localhost/GitHub/aria-practices/examples/landmarks/complementary.html');
+addLandmarkRole('contentinfo', false, 'Contentinfo Landmark', 'http://localhost/GitHub/aria-practices/examples/landmarks/contentinfo.html');
+addLandmarkRole('form', true, 'Form Landmark', 'http://localhost/GitHub/aria-practices/examples/landmarks/form.html');
+addLandmarkRole('main', true, 'Main Landmark', 'http://localhost/GitHub/aria-practices/examples/landmarks/main.html');
+addLandmarkRole('navigation', true, 'Navigation Landmark', 'http://localhost/GitHub/aria-practices/examples/landmarks/navigation.html');
+addLandmarkRole('region', true, 'Region Landmark', 'http://localhost/GitHub/aria-practices/examples/landmarks/region.html');
+addLandmarkRole('search', true, 'Search Landmark', 'http://localhost/GitHub/aria-practices/examples/landmarks/search.html');
+
+var exampleIndexFile = fs.readFileSync(fileNameTemplate, function (err) {
+  console.log('Error reading aria index:', err);
+});
+
+var sorted = [];
+
+for (role in indexOfRoles) {
+  sorted.push(role);
+}
+
+sorted.sort();
+
+html = '';
```

Comment:
In places where HTML is generated via string concatenation, consider using a JS template literal with some utility functions, such as

```javascript
const html = sorted.reduce((role, set) => {
  const examples = indexOfRoles[role];

  let examplesHTML = '';
  if (examples.length === 1) {
    examplesHTML = `<a href="${examples[0].ref}">${examples[0].title}</a>`;
  else {
    const exampleListItem = item => `<li><a href="${item.ref}">${item.title}</a></li>`;
    examplesHTML = `<ul>${examples.map(exampleListItem).join('\n')}</ul>`;
  }

  return set + `
    <tr>
      <td><code>${role}</code></td>
      <td>${examplesHtml}</td>
    </tr>
  `;
}, '');